.devcontainer/Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM mcr.microsoft.com/devcontainers/base:ubuntu-22.04
+
+ADD https://packages.microsoft.com/config/ubuntu/22.04/packages-microsoft-prod.deb /packages-microsoft-prod.deb
+ADD https://deb.nodesource.com/setup_20.x /nodesource_setup.sh
+
+ENV APT_PACKAGES="git curl zip unzip tar build-essential cmake ninja-build powershell nodejs"
```

Comment:
I'm pretty sure I need to add some more here for binary caching e2e tests to pass due to mono. Testing in GitHub Codespaces now